python/ray/tests/runtime_env_container/test_with_env_vars.py

```diff
@@ -0,0 +1,32 @@
+import argparse
+import os
+
+import ray
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--image", type=str, help="The docker image to use for Ray worker")
+parser.add_argument(
+    "--use-image-uri-api",
+    action="store_true",
+    help="Whether to use the new `image_uri` API instead of the old `container` API.",
+)
+args = parser.parse_args()
+
+if args.use_image_uri_api:
+    runtime_env = {
+        "image_uri": args.image,
+        "env_vars": {"TEST_DEF": "1"},
+    }
+else:
+    runtime_env = {
+        "container": {"image": args.image},
+        "env_vars": {"TEST_DEF": "hi world"},
+    }
+
+
+@ray.remote(runtime_env=runtime_env)
+def f():
+    return os.environ.get("TEST_DEF")
```

Comment:
can you also check

- regular env var is not inherited (e.g., set os.setenv in python script and make sure they are not inherited)
- RAY_env vars are inherited

Reply:
added